scheduler/crates/api/src/calendar/update_calendar.rs

```diff
@@ -100,21 +97,11 @@ impl UseCase for UpdateCalendarUseCase {
         };
 
         if let Some(wkst) = self.week_start {
-            if !calendar.settings.set_week_start(wkst) {
-                return Err(UseCaseError::InvalidSettings(format!(
-                    "Invalid week start: {}, must be between 0 and 6",
-                    wkst
-                )));
-            }
+            calendar.settings.week_start = wkst;
         }
 
-        if let Some(timezone) = &self.timezone {
-            if !calendar.settings.set_timezone(timezone) {
-                return Err(UseCaseError::InvalidSettings(format!(
-                    "Invalid timezone: {}, must be a valid IANA Timezone string",
-                    timezone
-                )));
-            }
+        if let Some(timezone) = self.timezone {
+            calendar.settings.timezone = timezone;
```

Comment:
This is a nice cleanup in the code. I chose initially to use `String` to represent `timezone` attribute on `RequestBody` so that the error message returned to the client could be customised, but on second though I think the error message returned by a deserialize error is good enough.

Reply:
Yep, that was also my opinion also. They are good enough.